Upstream-Status: Backport
Signed-off-by: Ross Burton <ross.burton@arm.com>

From 89958f7f30ec722e30e1bcffdeab547c874fa475 Mon Sep 17 00:00:00 2001
From: Adrian Herrera <adrian.herrera@arm.com>
Date: Mon, 15 Mar 2021 13:14:44 +0000
Subject: [PATCH] python: debug, fix Mapping import

Change "collections.Mapping" to "collections.abc.Mapping".
"collections.Mapping" was an alias, it is deprecated starting from Python 3.3, and it will be removed in Python 3.10.

Change-Id: Ic257e3c5206eb3d48d4eed85a93fac48bd3b8dc4
Signed-off-by: Adrian Herrera <adrian.herrera@arm.com>
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/43023
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>
---
 src/python/m5/debug.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/python/m5/debug.py b/src/python/m5/debug.py
index d808850cc..787a39ece 100644
--- a/src/python/m5/debug.py
+++ b/src/python/m5/debug.py
@@ -26,7 +26,7 @@

 from __future__ import print_function

-from collections import Mapping
+from collections.abc import Mapping

 import _m5.debug
 from _m5.debug import SimpleFlag, CompoundFlag
-- 
2.25.1

